console/src/tasks.rs

```diff
@@ -114,31 +138,67 @@ impl State {
         if let Some(now) = update.now {
             self.last_updated_at = Some(now.into());
         }
+
+        if let Some(new_metadata) = update.new_metadata {
+            let metas = new_metadata.metadata.into_iter().map(|meta| {
+                let id = meta.id.expect("no id").id;
+                let metadata = meta.metadata.expect("no metadata");
+                (id, metadata.into())
+            });
```

Comment:
nit/TIOLI: it's _probably_ better to skip any metadata that's empty (and maybe log a warning?), rather than using `expect` and panicking if it's empty. we could write this instead:
```suggestion
            let metas = new_metadata.metadata.into_iter().filter_map(|meta| {
                let id = meta.id?.id;
                let metadata = meta.metadata?;
                Some((id, metadata.into()))
            });
```
this will silently skip any metadata where the `id` or `metadata` field is none --- you could add logging as well.

the subscriber *should* never send us empty metadata, but if there's a bug in the subscriber implementation, or if we make backwards compatible proto changes and those fields become optional, it's probably better to keep running and skip them, rather than panicking?

not a big deal for now though.